feat(sdk): Add get_logs() method to download component logs

SoumyaRaikwar opened the pull request:
## What this PR does
Implements SDK method to download component logs from pipeline runs, addressing the feature request in #12388.

## Changes
- `get_logs()`: Public method to retrieve logs by run ID and optional component name
- `_get_workflow_name_from_run()`: Helper to extract Argo workflow names
- `_extract_component_name_from_pod()`: Helper to parse component names from pod metadata
- Comprehensive unit tests with mocking (no cluster required)

## Features
 Fetch single component logs (returns string)  
 Fetch all component logs (returns dict)  
 Multi-user namespace support  
 Graceful container fallback handling

## Testing
- [x] 11 new test cases
- [x] Code formatted with YAPF
- [x] Follows existing test patterns

Fixes #12388

SoumyaRaikwar committed:
feat(sdk): Add get_logs() method to download component logs

Implements #12388 by adding SDK support for downloading logs from
pipeline run components.

Changes:
- Added get_logs() public method to Client class
- Added helper methods _get_workflow_name_from_run() and
  _extract_component_name_from_pod()
- Added comprehensive unit tests with mocking
- Leverages existing kubernetes dependency (v30.1.0)

Features:
- Fetch logs from specific component by name
- Fetch logs from all components in a run
- Support for multi-user namespace configurations
- Graceful fallback for different container configurations

Tests:
- 11 new test cases covering success and error scenarios
- All tests use mocking (no cluster required)
- Follows existing test patterns from client_test.py

Fixes #12388

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

mprahl commented on sdk/python/kfp/client/client.py:
```diff
@@ -1360,6 +1360,129 @@ def get_run(self, run_id: str) -> kfp_server_api.V2beta1Run:
+
+        namespace = namespace or self.get_user_namespace()
+
+        run = self.get_run(run_id=run_id)
```
I'd rather this leverage the log artifacts automatically uploaded by the KFP launcher. This has the benefit of still working even if the pod gets deleted and it doesn't require the user to have access to the Kubernetes cluster itself.

SoumyaRaikwar committed:
feat(sdk): Use artifact-based approach for get_logs()

Changed implementation from Kubernetes pod querying to using KFP's
ReadArtifact API. This approach:
- Works even after pod deletion
- Doesn't require Kubernetes cluster access
- Leverages KFP's built-in artifact storage

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
Merge branch 'master' into feature/add-get-logs-sdk

SoumyaRaikwar committed:
fix(sdk): resolve CI failures in get_logs() implementation. Fixes #12388

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
fix(sdk): resolve presubmit failures, format strings, and add missing docstrings. Fixes #12388

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
fix(sdk): refactor _read_artifact to use direct REST API for log retrieval

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
chore(sdk): fix stylistic issues in client and test scripts

This resolves failing YAPF, isort, and docformatter checks in CI.

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
chore(sdk): resolve final stylistic issues and unused imports

This addresses the pycln and string_fixer failures in CI for the functional test script.

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
chore(sdk): remove functional test script from repository

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
chore(sdk): finalize stylistic fixes and import sorting

This resolves isort and string_fixer failures in CI by standardizing imports in
__init__.py and quote usage in client.py.

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
fix: format imports in kfp/client/__init__.py with isort

Signed-off-by: SoumyaRaikwar <somuraik@gmail.com>

SoumyaRaikwar committed:
Merge branch 'master' into feature/add-get-logs-sdk
